src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -1829,10 +1832,13 @@ methodHandle SharedRuntime::reresolve_call_site(TRAPS) {
                  iter.type() == relocInfo::opt_virtual_call_type
                 , "unexpected relocInfo. type");
         }
+        patch_call = true;
       } else {
-        assert(!UseInlineCaches, "relocation info. must exist for this address");
+        assert(!UseInlineCaches || caller_nm->is_compiled_by_jvmci(), "relocation info. must exist for this address");
```

Comment:
Could we please make this a `guarantee`. This is not a particularly hot code path and the failure it can cause later is very hard to debug.